tests/lib/ocpp/v201/test_database_handler.cpp

```diff
@@ -0,0 +1,77 @@
+
+#include "ocpp/v201/database_handler.hpp"
+#include "ocpp/v201/ocpp_types.hpp"
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <memory>
+
+namespace ocpp::v201 {
+
+class DatabaseHandlerV201 : public testing::Test {
+
+public:
+    DatabaseHandlerV201() {
+        std::unique_ptr<common::DatabaseConnection> dbc = std::make_unique<common::DatabaseConnection>("file::memory:?cache=shared");
+        dbc->open_connection(); // Open connection so memory stays shared
+
+        this->db_handler = std::make_unique<DatabaseHandler>(std::move(dbc),
+                                                             std::filesystem::path(MIGRATION_FILES_LOCATION_V201));
+        this->db_handler->open_connection();
+        this->db_connection = std::make_unique<common::DatabaseConnection>("file::memory:?cache=shared");
+        db_connection->open_connection(); // Open connection so memory stays shared
+    }
+    std::unique_ptr<common::DatabaseConnection> db_connection;
+    std::unique_ptr<DatabaseHandler> db_handler;
+
+protected:
+    void SetUp() override {
+    }
+};
+
+TEST_F(DatabaseHandlerV201, KO1_FR27_DatabaseWithNoData_InsertProfile) {
+    db_handler->insert_or_update_charging_profile(1, ChargingProfile{1,1});
+    std::string sql = "SELECT COUNT(*) FROM CHARGING_PROFILES";
+    auto select_stmt = db_connection->new_statement(sql);
+    
+    if (select_stmt->step() != SQLITE_ROW) {
+        // TODO ...
```

Comment:
For these TODOs, do you think we could add a little more context regarding what we should be putting here eventually?

Reply:
I added a check to make sure a row was returned. If it is not then no data was saved so we should fail.